src/lib.rs

```diff
@@ -107,9 +109,8 @@ where
 {
     buffer: &'a mut [u8],
     used: usize,
-    /// Maximum four levels deep
-    menus: [Option<&'a Menu<'a, T>>; 4],
-    depth: usize,
+    menu: menu_manager::MenuManager<'a, T>,
```

Comment:
```suggestion
    menu_mgr: menu_manager::MenuManager<'a, T>,
```

If it's the menu manager, let's call it that!